org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/ASTParser.java

```diff
@@ -1166,13 +1164,98 @@ public IBinding[] createBindings(IJavaElement[] elements, IProgressMonitor monit
 			if ((this.bits & CompilationUnitResolver.IGNORE_METHOD_BODIES) != 0) {
 				flags |= ICompilationUnit.IGNORE_METHOD_BODIES;
 			}
-			return CompilationUnitResolver.resolve(elements, this.apiLevel, this.compilerOptions, this.project, this.workingCopyOwner, flags, monitor);
+
+			return resolve(elements, this.apiLevel, this.compilerOptions, this.project, this.workingCopyOwner, flags, this.unitResolver, monitor);
 		} finally {
 			// reset to defaults to allow reuse (and avoid leaking)
 			initializeDefaults();
 		}
 	}
 
+
+	static IBinding[] resolve(
```

Comment:
As this is API, and `CompilationUnitResolver.resolve(..)` was API, should it not have had some kind of javadoc ? I know nearly all the methods in `CompilationUnitResolver` don't but most in `ASTParser` do.

Reply:
1. Also, I would just rename the commit title to `Ensure calls to ASTParser.createBindings use correct unit resolver` and mention `- Fixes #3428` in the body. Also can probably get rid of your extra "Cleanup" and Signed-off-by message.